src/Microsoft.Health.SqlServer/Features/Health/SqlServerHealthCheck.cs

```diff
@@ -20,23 +22,32 @@ public class SqlServerHealthCheck : IHealthCheck
     {
         private readonly ILogger<SqlServerHealthCheck> _logger;
         private readonly SqlConnectionWrapperFactory _sqlConnectionWrapperFactory;
+        private readonly IAsyncPolicy _retrySqlPolicy;
 
-        public SqlServerHealthCheck(SqlConnectionWrapperFactory sqlConnectionWrapperFactory, ILogger<SqlServerHealthCheck> logger)
+        public SqlServerHealthCheck(
+            SqlConnectionWrapperFactory sqlConnectionWrapperFactory,
+            ISqlServerTransientFaultRetryPolicyFactory sqlServerTransientFault,
+            ILogger<SqlServerHealthCheck> logger)
         {
             _sqlConnectionWrapperFactory = EnsureArg.IsNotNull(sqlConnectionWrapperFactory, nameof(sqlConnectionWrapperFactory));
+            EnsureArg.IsNotNull(sqlServerTransientFault, nameof(sqlServerTransientFault));
+            _retrySqlPolicy = sqlServerTransientFault.Create();
             _logger = EnsureArg.IsNotNull(logger, nameof(logger));
         }
 
         public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken)
         {
             try
             {
-                using SqlConnectionWrapper sqlConnectionWrapper = await _sqlConnectionWrapperFactory.ObtainSqlConnectionWrapperAsync(cancellationToken);
-                using SqlCommandWrapper sqlCommandWrapper = sqlConnectionWrapper.CreateSqlCommand();
+                await _retrySqlPolicy.ExecuteAsync(async () =>
+                {
+                    using SqlConnectionWrapper sqlConnectionWrapper = await _sqlConnectionWrapperFactory.ObtainSqlConnectionWrapperAsync(cancellationToken);
+                    using SqlCommand sqlCommand = sqlConnectionWrapper.SqlConnection.CreateCommand();
 
-                sqlCommandWrapper.CommandText = "select @@DBTS";
+                    sqlCommand.CommandText = "select @@DBTS";
 
-                await sqlCommandWrapper.ExecuteScalarAsync(cancellationToken).ConfigureAwait(false);
+                    await sqlCommand.ExecuteScalarAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
Should we instead embed the retry policy within the `SqlCommandWrapper`? Perhaps in like a `RetryableSqlCommand` or something where `ExecuteScalerAsync` uses the policy under the hood? That way each caller doesn't need to explicitly use the Polly retry policy?